elide-core/src/main/java/com/yahoo/elide/parsers/expression/FilterExpressionVisitor.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2016, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.parsers.expression;
+
+import com.yahoo.elide.core.CheckInstantiator;
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.filter.Operator;
+import com.yahoo.elide.core.filter.Predicate;
+import com.yahoo.elide.core.filter.expression.AndFilterExpression;
+import com.yahoo.elide.core.filter.expression.FilterExpression;
+import com.yahoo.elide.core.filter.expression.NotFilterExpression;
+import com.yahoo.elide.core.filter.expression.OrFilterExpression;
+import com.yahoo.elide.generated.parsers.ExpressionBaseVisitor;
+import com.yahoo.elide.generated.parsers.ExpressionParser;
+import com.yahoo.elide.security.FilterExpressionCheck;
+import com.yahoo.elide.security.RequestScope;
+import com.yahoo.elide.security.checks.Check;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/*
+FilterExpressionVisitor parses a permission parseTree and returns the corresponding Predicate representation of it.
+ */
+public class FilterExpressionVisitor extends ExpressionBaseVisitor<FilterExpression> implements CheckInstantiator {
+    private final EntityDictionary dictionary;
+    private final RequestScope requestScope;
+
+    public FilterExpressionVisitor(EntityDictionary dictionary,RequestScope requestScope) {
+        this.dictionary = dictionary;
+        this.requestScope = requestScope;
+    }
+
+    @Override
+    public FilterExpression visitNOT(ExpressionParser.NOTContext ctx) {
+        // Create a not expression
+        return new NotFilterExpression(visit(ctx.expression()));
+    }
+
+    @Override
+    public FilterExpression visitOR(ExpressionParser.ORContext ctx) {
+        return new OrFilterExpression(visit(ctx.left), visit(ctx.right));
+    }
+
+    @Override
+    public FilterExpression visitAND(ExpressionParser.ANDContext ctx) {
+        FilterExpression left = visit(ctx.left);
+        FilterExpression right = visit(ctx.right);
```

Comment:
You should either make local variables like this or do things in line, but most importantly you should be consistent.